crates/futures/src/lib.rs

```diff
@@ -103,300 +103,14 @@
 
 #![deny(missing_docs)]
 
-use std::cell::{Cell, RefCell};
-use std::fmt;
-use std::rc::Rc;
-use std::sync::Arc;
+#[cfg(feature = "nightly")]
+mod nightly;
 
-use futures::executor::{self, Notify, Spawn};
-use futures::future;
-use futures::prelude::*;
-use futures::sync::oneshot;
-use js_sys::{Function, Promise};
-use wasm_bindgen::prelude::*;
+#[cfg(feature = "nightly")]
+pub use nightly::*;
 
-/// A Rust `Future` backed by a JavaScript `Promise`.
-///
-/// This type is constructed with a JavaScript `Promise` object and translates
-/// it to a Rust `Future`. This type implements the `Future` trait from the
-/// `futures` crate and will either succeed or fail depending on what happens
-/// with the JavaScript `Promise`.
-///
-/// Currently this type is constructed with `JsFuture::from`.
-pub struct JsFuture {
-    resolved: oneshot::Receiver<JsValue>,
-    rejected: oneshot::Receiver<JsValue>,
-    callbacks: Option<(Closure<FnMut(JsValue)>, Closure<FnMut(JsValue)>)>,
-}
+#[cfg(not(feature = "nightly"))]
```

Comment:
I think that we'll probably want to have the standard library futures support in its own module. We don't want enabling the `nightly` feature to break this crate (features should be additive in terms of functionality) for existing users. That does mean that if `nightly` is enabled it'll pull in two implementations of futures, but I think that's ok since the linker will eliminate them if they're not used.

Reply:
I'm okay with that.